### ISSUE TITLE
Fable cli options are case sensitive but allows case insensitive options

### ISSUE THREAD

**Issue by kjuulh** [fable-compiler/Fable#2561]
### Description

When using fable from dotnet tools it doesn't provide a hard failure on a wrong input option i.e. `dotnet tool fable src --outdir` vs `dotnet tool fable src --outDir`. This will work it will transpile to the javascript files, but not in the output dir. I haven't tested whether it is an issue only with outdir, or if it is also the case for other options. 

This may be a design decision. However, for me this seemed unintuitive. Feel free to close this issue if it is intended behavior.

Edit: Can confirm this seems to be intended behavior. `dotnet fable src --123` Finished a restore and stuff successfully.

This is my first time submitting an issue to this repo, so please be gentle =D

### Repro code

```bash
dotnet new -i Fable.Template
dotnet new fable
dotnet tool restore

dotnet fable src --sourceMap --outdir public/

ls public/
```

### Expected and actual results

Actual
fable.ico  index.html

Expected (error)
--outdir is not an option

Expected (case insensitive)
fable.ico  index.html App.fs.js

### Related information

* Fable version: `dotnet fable --version`
  * 3.2.9
* Operating system
  * Linux Manjaro

**Comment by MangelMaxime** [comment]
What's happening here is that Fable is not strict about the options you pass to it.

So when there are options it doesn't know, it just ignore them and run as if they were not provided.

@alfonsogarciacaro Perhaps, it could be a good idea to use a CLI parser library here instead of making our own parser. I think Argu is pretty popular in F#

**Comment by alfonsogarciacaro** [comment]
At the beginning it was to keep things simple and avoid dependencies. We have some other dependencies now so I wouldn't mind to add Argu (though it's not [without quicks](https://github.com/Zaid-Ajaj/Femto/pull/78) either), but it also entails changing the code so I thought a simple solution would be better... just to touch a lot of code myself 😅 Anyways, should be fixed by #2562. I will push a new release.
